enhancements/sig-architecture/227-manifestworkreplicaset-rollout-plugin/README.md

```diff
@@ -0,0 +1,378 @@
+* Total Cluster Count: The total number of clusters selected by the current Placement.
+* Rollout Status: The cluster rollout status (completed clusters, progressing clusters, timed_out clusters, removed clusters). Each entry in the list includes:
+  - clusterName: The name of the cluster.
+  - rolloutStatus: The current [cluster rollout status](https://github.com/open-cluster-management-io/sdk-go/blob/main/pkg/apis/cluster/v1alpha1/rollout.go#L23-L39) (e.g., ToApply, Progressing, Succeeded, Failed, TimeOut, Skip).
+  - manifestRevisionName: The name of the manifest revision applied to the cluster.
+
+### Configure custom plugin for work controller
+
+A new `workConfiguration.plugin` field is introduced in ClusterManager to configure the rollout plugin:
+
+```yaml
+apiVersion: operator.open-cluster-management.io/v1
+kind: ClusterManager
+metadata:
+  name: cluster-manager
+spec:
+  ...
+  workImagePullSpec: quay.io/open-cluster-management/work:v1.0.0
+  workConfiguration:
+    workDriver: kube
+    # Plugin configuration
+    plugin:
+      # the image name of plugin sidecar
+      image: quay.io/open-cluster-management/my-rollout-plugin:v0.0.1
+      # localhost sidecar port
+      # port: 6767
+```
+
+#### Deployment manifest change to use plugin
+
+To run plugin server as a sidecar, [deployment manifest](https://github.com/open-cluster-management-io/ocm/blob/main/manifests/cluster-manager/management/work/deployment.yaml) will include `initContainers` field to run plugin image.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+...
+spec:
+  ...
+  selector:
+    ...
+  template:
+      ...
+    spec:
+      ...
+      initContainers:
+      - name: {{ .ClusterManagerName }}-plugin
+        image: {{ .PluginImage }}
+        restartPolicy: Always
+        args:
+          - "/plugin"
+        env:
+          - name: PLUGIN_SERVER_PORT
+            value: {{ .PluginServerPort }}
+      containers:
+      - name: {{ .ClusterManagerName }}-work-controller
+        image:  {{ .WorkImage }}
+        ...
+        env:
+        {{- if .PluginServerPort }}
+          - name: PLUGIN_SERVER_PORT
+            value: {{ .PluginServerPort }}
+        {{- end }}
+```
+
+### Open Questions [optional]
+
+> WIP
+
+### Test Plan
+
+- Unit-test
+- Integration-test: Create the sample K8s deployment based safe rollout plugin.
+
+### Graduation Criteria
+
+> TBD
+
+### Upgrade / Downgrade Strategy
+
+The plugin mechanism is opt-in. Work controller without plugin configuration behave identically to today’s work controller. Additive CRD changes (new condition PluginLoaded) are backward-compatible.
+
+### Version Skew Strategy
+
+The proposed changes introduce new condition and new fields in the exising custom resources. 
+
+## Alternatives
+
+- Implement plugin server as a standalone service: In order to run plugin server securely, we need to enable TLS connection. Using a sidecar, we can avoid the complex network and security configuration.
```

Comment:
on the other hand, plugin can be easily customized.

Reply:
It might be better to run standalone services. Do you prefer running plugin service as a standalone service? @qiujian16